{"description": "Versions 1.17 and 1.18 of the Python urllib3 library suffer from a vulnerability that can cause them, in certain configurations, to not correctly validate TLS certificates. This places users of the library with those configurations at risk of man-in-the-middle and information leakage attacks. This vulnerability affects users using versions 1.17 and 1.18 of the urllib3 library, who are using the optional PyOpenSSL support for TLS instead of the regular standard library TLS backend, and who are using OpenSSL 1.1.0 via PyOpenSSL. This is an extremely uncommon configuration, so the security impact of this vulnerability is low.", "methods": ["Uses a vulnerable library (urllib3) to bypass certificate validation.", "Uses a non-standard TLS backend (PyOpenSSL) to bypass certificate validation.", "Uses a specific version of OpenSSL (1.1.0) to bypass certificate validation."]}